test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -1281,6 +1293,56 @@ public void TensorFlowImageClassification()
                 Assert.Equal(1, metrics.MicroAccuracy);
                 Assert.Equal(1, metrics.MacroAccuracy);
             }
+
+            // Testing TrySinglePrediction: Utilizing PredictionEngine for single
+            // predictions.
+            var predictionEngine = mlContext.Model
+                .CreatePredictionEngine<ImageData, ImagePrediction>(loadedModel);
+
+            IEnumerable<ImageData> testImages = LoadImagesFromDirectory(
+                fullImagesetFolderPath, true);
+
+            string[] directories = Directory.GetDirectories(fullImagesetFolderPath);
+            string[] labels = new string[directories.Length];
+            for(int j = 0; j < labels.Length; j++)
+            {
+                var dir = new DirectoryInfo(directories[j]);
+                labels[j] = dir.Name;
+            }
```

Comment:
this is problematic, you cannot assume the ValueToKey transform will see the directories as labels in this order. Instead get keys from the IDataview like we were doing in the sample before 

VBuffer<ReadOnlyMemory<char>> keys = default;
                loadedModel.GetOutputSchema(schema)["Label"].GetKeyValues(ref keys);